src/g3log/g3log.hpp

```diff
@@ -145,17 +145,17 @@ namespace g3 {
 
 
 // LOG(level) is the API for the stream log
-#define LOG(level) if(!g3::logLevel(level)) {} else INTERNAL_LOG_MESSAGE(level).stream()
+#define LOG(level) if (!g3::logLevel(level)) {} else INTERNAL_LOG_MESSAGE(level).stream()
 
 
 // 'Conditional' stream log
 #define LOG_IF(level, boolean_expression)  \
-   if (false == (boolean_expression) || !g3::logLevel(level)) {} else INTERNAL_LOG_MESSAGE(level).stream()
+   if (!g3::logLevel(level) || !(boolean_expression)) {} else INTERNAL_LOG_MESSAGE(level).stream()
```

Comment:
Revert these changes. Although C++ allows a ptr as a valid “Boolean” it’s poor practice. 
I prefer this code to be as is

Reply:
`boolean_expression` checking is reverted. As for checking if log level is enabled before evaluating the expression, I still believe it's necessary.